File: Website/AdminSite/models/movie.js
export class Movie {
  constructor(
    title,
    duration,
    description,
    releaseDate,
    director,
    actor,
    poster,
    trailer,
    country,
    language,
    basePrice,
    categories,
    classify
  ) {
    this.title = title;
    this.duration = duration;
    this.description = description;
    this.releaseDate = releaseDate;
    this.director = director;
    this.actor = actor;
    this.poster = poster;
    this.trailer = trailer;
    this.country = country;
    this.language = language;
    this.basePrice = basePrice;
    this.categories = categories;
    this.classify = classify;
  }
}

File: App/movie_booking_app/lib/models/movie/movie.dart
import 'package:movie_booking_app/models/category/categories.dart';

class Movie {
  final int id;
  final String title;
  final String classify;
  final List<Categories> categories;
  final String poster;
  final DateTime? releaseDate;
  final bool isRelease;
  final String? trailer;
  Movie(
      {required this.isRelease,
      required this.id,
      required this.title,
      required this.classify,
      required this.categories,
      required this.poster,
      required this.trailer,
      this.releaseDate});

  factory Movie.fromJson(Map<String, dynamic> json) {
    List<dynamic> categories = json['categories'];
    List<Categories> listCategory = categories.map((category) {
      return Categories(
        id: category['id'],
        name: category['name'],
      );
    }).toList();
    return Movie(
      id: json['id'],
      title: json['title'],
      classify: json['classify'],
      categories: listCategory,
      poster: json['poster'],
      trailer: json['trailer'],
      isRelease: json['release'],
    );
  }
}
